adafruit_onewire/bus.py

```diff
@@ -0,0 +1,206 @@
+        :param int end: Index to write up to but not include
+        """
+        start = 0
+        end = len(buf)
+        if 'start' in kwargs:
+            start = kwargs['start']
+        if 'end' in kwargs:
+            end = kwargs['end']
+        for i in range(start, end):
+            buf[i] = self._readbyte()
+
+    def write(self, buf, **kwargs):
+        """
+        Write the bytes from ``buf`` to the device.
+
+        If ``start`` or ``end`` is provided, then the buffer will be sliced
+        as if ``buffer[start:end]``. This will not cause an allocation like
+        ``buffer[start:end]`` will so it saves memory.
+
+        :param bytearray buf: buffer containing the bytes to write
+        :param int start: Index to start writing from
+        :param int end: Index to read up to but not include
+        """
+        start = 0
+        end = len(buf)
+        if 'start' in kwargs:
+            start = kwargs['start']
+        if 'end' in kwargs:
+            end = kwargs['end']
+        for i in range(start, end):
+            self._writebyte(buf[i])
+
+    def scan(self):
+        """Scan for devices on the bus and return a list of addresses."""
+        devices = []
+        diff = 65
+        rom = False
+        for i in range(0xff): # pylint: disable=unused-variable
+            rom, diff = self._search_rom(rom, diff)
+            if rom:
+                devices += [OneWireAddress(rom)]
+            if diff == 0:
+                break
+        return devices
+
+    def _readbit(self):
+        return self._ow.read_bit()
+
+    def _readbyte(self):
+        val = 0
+        for i in range(8):
+            val |= self._ow.read_bit() << i
+        return val
+
+    def _writebit(self, value):
+        return self._ow.write_bit(value)
+
+    def _writebyte(self, value):
+        for i in range(8):
+            bit = (value >> i) & 0x1
+            self._ow.write_bit(bit)
+        return
+
+    def _search_rom(self, l_rom, diff):
+        if not self.reset():
+            return None, 0
+        self._writebyte(_SEARCH_ROM)
+        if not l_rom:
+            l_rom = bytearray(8)
+        rom = bytearray(8)
+        next_diff = 0
+        i = 64
+        for byte in range(8):
+            r_b = 0
+            for bit in range(8):
+                b = self._readbit()
+                if self._readbit():
+                    if b: # there are no devices or there is an error on the bus
+                        return None, 0
+                else:
+                    if not b: # collision, two devices with different bit meaning
+                        if diff > i or ((l_rom[byte] & (1 << bit)) and diff != i):
+                            b = 1
+                            next_diff = i
+                self._writebit(b)
+                if b:
+                    r_b |= 1 << bit
+                i -= 1
+            rom[byte] = r_b
+        return rom, next_diff
+
+    def crc8(self, data): # pylint: disable=no-self-use
```

Comment:
Instead of disabling pylint, this can be a static method. That means it works independent of the actual object state. To do so, use the `@staticmethod` decorator and drop `self`.

```python
@staticmethod
def crc8(data):
```